lib/spack/spack/test/conftest.py

```diff
@@ -1138,3 +1144,16 @@ def clear_directive_functions():
     # proceeding with subsequent tests that may depend on the original
     # functions.
     spack.directives.DirectiveMeta._directives_to_be_executed = []
+
+
+#########
+# Fixture to cache module function
+# The test is responsible for mocking it
+#########
+@pytest.fixture
+def save_module_func():
+    old_func = spack.util.module_cmd.module
+
+    yield
+
+    spack.util.module_cmd.module = old_func
```

Comment:
This looks like it can be substituted by the use of `monkeypatch.setattr` in the test function, unless I'm missing something.